x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -1078,6 +1093,95 @@ static String runEsqlAsTextWithFormat(RequestObjectBuilder builder, String forma
         return Streams.copyToString(new InputStreamReader(entity.getContent(), StandardCharsets.UTF_8));
     }
 
+    static String runEsqlAsyncAsTextWithFormat(RequestObjectBuilder builder, String format, @Nullable Character delimiter)
```

Comment:
There's now a `runEsqlSync()` (or even a `runEsql()` which takes a mode), it'd be worth refactoring that go get out a result.